### PR TITLE
TBEL: new methods

### PR DESCRIPTION
**Pull request by nickAS21**
#### new methods for downlink
- intToHex
- longToHex
- intLongToString
- floatToHex
- doubleToHex
- printUnsignedBytes (Example: print value=-40 to value=216; value=-1 to value=255)
- newError
- isBinary
- isOctal
- isDecimal
- isHexadecimal

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
 
## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [x] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java [hunk 1]
```diff
@@ -46,11 +46,23 @@
...
 
+    private static final int zeroRadix = 0;
+    private static final int octalRadix = 8;
+    private static final int decRadix = 10;
+    private static final int hexRadix = 16;
+    private static final int hexLenMin = -1;
+    private static final int hexLenIntMax = 8;
+    private static final int hexLenLongMax = 16;
+    private static final int bytesLenLongMax = 8;
+
```
Comment (ViacheslavKlimov) [inline common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:65]: Use upper-case for constants naming.

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java [hunk 2]
```diff
@@ -332,25 +389,96 @@ public static Long parseLong(String value, int radix) {
...
             try {
-                return Float.parseFloat(prepareNumberString(value));
+                if (radixValue == decRadix) {
+                    return Float.parseFloat(value);
+                } else {
+                    int bits = Integer.parseUnsignedInt(valueP, hexRadix);
+                    return Float.intBitsToFloat(bits);
+                }
             } catch (NumberFormatException e) {
+                throw new NumberFormatException(e.getMessage());
```
Comment (ViacheslavKlimov) [inline common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:458]: Why do you catch and throw the same exception?

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java [hunk 3]
```diff
@@ -419,39 +550,158 @@ public static double parseHexToDouble(String hex) {
...
+        String hex = Long.toHexString(number).toUpperCase();
+        hexLenMax = hexLenMax < 0 ? hex.length() : hexLenMax;
+        String hexWithoutZeroFF = removeLeadingZero_FF(hex, number, hexLenMax);
+        hexWithoutZeroFF = bigEndian ? hexWithoutZeroFF : reverseHexStringByOrder(hexWithoutZeroFF);
+        len = len == hexLenMin ? hexWithoutZeroFF.length() : len;
+        String result = hexWithoutZeroFF.substring(hexWithoutZeroFF.length() - len);
+        return pref ? "0x" + result : result;
+    }
+
+    private static String removeLeadingZero_FF(String hex, Long number, int hexLenMax) {
```
Comment (ViacheslavKlimov) [inline common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:662]: What does FF mean?

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java [hunk 4]
```diff
@@ -647,6 +897,15 @@ public static String decodeURI(String uri) {
...
     }
 
+    public static void newError(String message) {
+        newError(message, null);
+    }
+
+    public static void newError(String message, Object value) {
+        String msg = value == null ? message : message + " Value = " + value;
+        throw new RuntimeException(msg);
+    }
```
Comment (ViacheslavKlimov) [inline common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:907]: What is the point of these methods?

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java [hunk 5]
```diff
@@ -647,6 +897,15 @@ public static String decodeURI(String uri) {
         return URLDecoder.decode(uri, StandardCharsets.UTF_8);
     }
 
+    public static void newError(String message) {
+        newError(message, null);
+    }
+
+    public static void newError(String message, Object value) {
+        String msg = value == null ? message : message + " Value = " + value;
```
Comment (ViacheslavKlimov) [inline common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:905]: Consider renaming the value part to " for value X"

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java [hunk 6]
```diff
@@ -647,6 +897,15 @@ public static String decodeURI(String uri) {
         return URLDecoder.decode(uri, StandardCharsets.UTF_8);
     }
 
+    public static void newError(String message) {
```
Comment (ViacheslavKlimov) [inline common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:900]: Maybe "raiseError"?

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java (hunk 3 above)
Comment (nickAS21) [reply common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:662]: Example if number negative

Comment (nickAS21) [reply common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:662]: Zero "00"
00 00 00 00

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java (hunk 1 above)
Comment (nickAS21) [reply common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:65]: Ok

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java (hunk 2 above)
Comment (nickAS21) [reply common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:458]: Ok

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java (hunk 4 above)
Comment (nickAS21) [reply common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:907]: OK

File: common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java (hunk 5 above)
Comment (nickAS21) [reply common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbUtils.java:905]: ok

**Pull request merged**